### PR TITLE
Fix job image override

### LINKED ISSUE #76: Unable to override image for the upgrade job
**Describe the bug**
Unable to override image for the upgrade job.

**Expected behavior**
Able to override image for the upgrade job

**To Reproduce**
No way of overriding jobs in https://github.com/knative-sandbox/operator/blob/master/pkg/reconciler/common/images.go#L46

**Knative release version**
0.14

### PR DESCRIPTION
**Pull request by aliok**
Fixes #76 

## Proposed Changes

* Fix image override for jobs (eg. 0.14 upgrade job for eventing)

**Release Note**

```release-note
Fixed image override for Knative Eventing 0.14 upgrade job
```

### CODE REVIEW

File: pkg/reconciler/common/images_test.go [hunk 1]
```diff
@@ -304,6 +311,13 @@ func validateUnstructedDaemonSetChanged(t *testing.T, tt *updateImageTest, u *un
 	util.AssertDeepEqual(t, daemonSet.Spec.Template.Spec.Containers, tt.expected)
 }
 
+func validateUnstructedJobChanged(t *testing.T, tt *updateImageTest, u *unstructured.Unstructured) {
```
Comment (houshengbo) [inline pkg/reconciler/common/images_test.go:314]: @aliok Would you mind doing a search for `unstructed` across all files. I think we should use `Unstructured`.

Comment (aliok) [reply pkg/reconciler/common/images_test.go]: Good catch :) I did a search&replace now in the entire codebase

**Pull request merged**